client/fenix/event.py

```diff
@@ -0,0 +1,17 @@
+"""Work with different event types"""
+
+import enum
+
+class Event(enum.Enum):
+    """
+    Enumeration of all events. The ordering should not be relied on.
+
+    CONN_FAIL indicates a failure to connect to the Fenix server
+    MSG_RECV indicates an incoming message event
+    MSG_SEND indicates an outgoing message event
+    QUIT should be used to terminate all threads cleanly
+    """
+    CONN_FAIL = enum.auto()
+    MSG_RECV = enum.auto()
+    MSG_SEND = enum.auto()
```

Comment:
In Fenix server I had planned on the JSON field `type` recording the message type and a `data` or `payload` field containing the rest of the data.  Also, unless seperately you convert these to `Event` then we are depending on the ordering, as enum assigns a int iirc.

I would fix this by having an event type that is `WEBSOCKET_EVENT` (or similar) and is dispatched in the queue when a data is sent over the WebSocket, and then process individual WebSocket event types later.

Reply:
This would couple the internal logic to the API specification, which is harder to update when the API changes, and harder to test.

Reply:
What internal logic specifically?  You need to be able to process API requests.  Do it how you want, but you *are* depending on ordering of your enum for the API, and also know that current plan for the API is in the following form, unless you have a better idea.

```json
{
    "type": "sendmessage",
    "payload": "{...}",
}
```

Reply:
I'm planning to convert each JSON event to an (Event, payload) pair. I see your point, and it might be refactored later, but as is it's most convenient not to be working with a specific JSON format.